arangod/RestServer/LanguageCheckFeature.cpp

```diff
@@ -169,6 +169,20 @@ std::tuple<std::string, LanguageType> getOrSetPreviousLanguage(
 
   return {std::string{collatorLang}, currLangType};
 }
+
+bool compareLangWithPrev(std::string const& lang, std::string const& prevLang) {
+  if (lang == prevLang) {
+    return true;
+  }
+  // Earlier ICU versions (for example in ArangoDB <= 3.11) derived different
+  // values for the "C" locale. For upgrades to work, we have to accept these
+  // old settings, too:
+  if (lang == "en_US_POSIX" && (prevLang == "en_US" || prevLang == "")) {
```

Comment:
I understand that `en_US_POSIX` and `en_US` should be treated identically.
But should other locales with a `_POSIX` suffix also compare equally to their non-posix counterparts, e.g. `de_DE_POSIX` and `de_DE`?

Reply:
As far as I see, a locale setting with `de_DE` or anything (and in particular also `de_DE_POSIX`  is always shortened to "de" with respect to what goes into the LANGUAGE file. Therefore, this is not necessary. And I would like to restrict the check to the bare necessary minimum.